Maxar-Corp/UMS-flowable-spring-boot
# Feature/simple happy path workflow

# Description

The main changes covered in this document relate to the new BPMN process definition. `ums_simple_happy_path.bpmn2.xml`

<img width="1454" alt="image" src="https://user-images.githubusercontent.com/84722754/163623118-8aa75348-faf7-400f-a0a3-d5ad6f1d6728.png">
This diagram can be viewed via the intellij flowable extension, bpmn.io, or through the local flowable-ui tomcat server.

## Type of change


# How Has This Been Tested?

Testing is missing :|

# Testing steps
- running through the postman requests found in `/UMS-flowable-spring-boot/flowable_simple_workflow.postman_collection.json` collection
1. Create the appropriate user
2. Start the workflow
3. From here, you must complete the following user tasks in this order
  a. Requester POSTs a usertask with an assignee value, status can be left empty
  b. Assignee posts a usertask with status set to "in progress"
  c. Assignee posts a usertask with status set to "to review"
  d. Requester POSTS a usertake with the status set to "complete"
4. The process should be done at this point, evaluate that the process is complete by checking the /processes endpoint
5. additional, you can test each user task for the assignee by not providing the expected status value and confirming that the workflow does NOT progress to the next step.
TODO: update code to check for missing assignee value, currently it just defaults to the requester if an assignee is not specified.
